src/hooks/use-database.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import { useContext } from 'react'
+import { DatabaseContext } from '../DatabaseProvider'
+import Database from '../Database'
+import invariant from '../utils/common/invariant'
+
+import type DatabaseType from '../Database'
+
+export function useDatabase(): DatabaseType {
+  const database = useContext(DatabaseContext)
+  invariant(
+    typeof database !== typeof Database,
```

Comment:
```suggestion
    database
```

I think it's safe to assume DatabaseContext will give you a Database or undefined